### PR TITLE
Auth : add support for delete user and send resetting password email

### PR DESCRIPTION
**Pull request by gcharbonnier**

### CODE REVIEW

**Comment by greenfield932** [comment]
What will happen if user was signed in and reset password command sent at the same time? Is current sign in state still valid?

**Comment by gcharbonnier** [comment]
I don’t see any mechanism in place to detect this state and automatically sign out already signed in when password has been reseted (but I don’t see neither any use case where this would be needed). 
So I assume, that any user already signed in remains signed in even when password gets reseted…

**Comment by greenfield932** [comment]
ok, seems ok for me, waiting for Lars to review/approve.

File: src/qtfirebaseauth.cpp [hunk 1]
```diff
@@ -217,6 +247,16 @@ void QtFirebaseAuth::onFutureEvent(QString eventId, firebase::FutureBase future)
...
         }
+        else if(eventId == __QTFIREBASE_ID + QStringLiteral(".auth.deleteUser"))
+        {
+            qDebug() << this << "::onFutureEvent Delete user successfully";
+            setSignIn(false);
+        }
+
+        else if(eventId == __QTFIREBASE_ID + QStringLiteral(".auth.resetEmail"))
+        {
+            qDebug() << this << "::onFutureEvent reset email sent successfully";
```
Comment (Larpon) [inline src/qtfirebaseauth.cpp]: Maybe add a signal here to let the programmer sign-out any sessions if he need to.
Signal name suggestion: `passwordResetEmailSent`
(So in QML the signal implementation name would be `onPasswordResetEmailSent`)

**Comment by Larpon** [comment]
@gcharbonnier - your code looks good to me - thanks for contributing! :)

Regarding the login/logout upon password reset - I think this is an individual app policy thing - in which QtFirebase shouldn't interfere. I can't read from the docs what the Firebase API does - but it looks like your session is still valid (it's probably similar to a regular Google account - if I recall correctly your session is valid until you click the link in the password reset mail).

So how about [adding a signal](https://github.com/Larpon/QtFirebase/pull/51/files#diff-140618d9beaa021cbb70f54d9d809a2eR258) `passwordResetEmailSent` - that's called when the Future completes successfully. Then let the programmer decide if he need to; logout the user or just let the session run on firebase default premises?

**Pull request merged**